lib/charms/finos_legend_libs/v0/legend_operator_testing.py

```diff
@@ -382,6 +382,47 @@ def _test_update_config_gitlab_relation(self, _container_stop_mock, _container_r
         expected_rel_data = {'legend-gitlab-redirect-uris': json.dumps(fake_callback_uris)}
         self.assertDictEqual(expected_rel_data, relation_data)
 
+    @mock.patch("ops.testing._TestingPebbleClient.restart_services")
+    @mock.patch("ops.testing._TestingPebbleClient.stop_services")
+    def _test_update_config_gitlab_relation_without_being_leader(self, _container_stop_mock, _container_restart_mock):
+        self.harness.begin_with_initial_hooks()
+
+        # Setup the URI getter mock,
+        mock_get_uris = self.patch(self.harness.charm, '_get_legend_gitlab_redirect_uris')
+        fake_callback_uris = ['legendary.callback.url']
+        mock_get_uris.return_value = fake_callback_uris
```

Comment:
Since the expectation is that these will never be called, there's no point in setting this return value.